controllers/nodehealthcheck_controller_test.go

```diff
@@ -446,6 +446,55 @@ var _ = Describe("Node Health Check CR", func() {
 					Expect(underTest.Status.UnhealthyNodes).To(BeEmpty())
 				})
 			})
+
+			When("two NHC CRs with different templates target the same unhealthy node", func() {
+
+				otherTestCRName := "other-test"
+				var otherTestCR *v1alpha1.NodeHealthCheck
+
+				BeforeEach(func() {
+					// prepare other CR but do not create yet, in order to have a predictable order of things to happen
+					otherTestCR = underTest.DeepCopy()
+					otherTestCR.SetName(otherTestCRName)
+					otherTestCR.Spec.RemediationTemplate = &v1.ObjectReference{
+						Kind:       "Metal3RemediationTemplate",
+						Namespace:  MachineNamespace,
+						Name:       "ok",
+						APIVersion: "test.medik8s.io/v1alpha1",
+					}
+
+					setupObjects(1, 2, true)
+				})
+
+				It("only one NHC should remediate that node", func() {
+
+					By("Verifying node is remediated by 1st NHC")
+					Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+					Expect(underTest.Status.InFlightRemediations).To(HaveLen(1))
+					Expect(underTest.Status.UnhealthyNodes).To(HaveLen(1))
+					Expect(underTest.Status.UnhealthyNodes[0].Remediations).To(HaveLen(1))
+
+					By("Creating a 2nd NHC")
+					Expect(k8sClient.Create(context.Background(), otherTestCR)).To(Succeed())
+					DeferCleanup(func() {
+						Expect(k8sClient.Delete(context.Background(), otherTestCR)).To(Succeed())
+					})
+					By("Verifying node is NOT remediated by 2nd NHC")
+					// wait for unhealthy node
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(otherTestCR), otherTestCR)).To(Succeed())
+						g.Expect(otherTestCR.Status.UnhealthyNodes).To(HaveLen(1))
+					}, "5s", "1s").Should(Succeed(), "unhealthy node not detected")
+					// ensure no remediation starts
+					Consistently(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(otherTestCR), otherTestCR)).To(Succeed())
+						g.Expect(otherTestCR.Status.InFlightRemediations).To(HaveLen(0))
```

Comment:
Nit: I think that in both cases (1st & 2nd NHC) it would be a good idea to check for the existence/absence of the  actual remediation CR and not just NHC status

Reply:
The alignment of status and CR existence is covered in several other test cases, so just for simplicity I keep it as it is. But thanks!